### PR TITLE
Indicator Fields Restructure

### PR DESCRIPTION
**Pull request by Ni-Knight**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/40509
fixes: https://github.com/demisto/etc/issues/40462
fixes: https://github.com/demisto/etc/issues/32701
related: https://github.com/demisto/content/pull/18051
related: https://github.com/demisto/content-docs/pull/1039

## Description
Restructure of Indicator fields, layouts and types mapping

Documentation:
Doc - https://docs.google.com/document/d/1Fm4HEOaqKQO5MnmAJPjHhAA0MdcxUu99T2F6jpmJbE8/edit?usp=sharing
PDF - [Indicator fields changes.pdf](https://github.com/demisto/content/files/8148669/Indicator.fields.changes.pdf)

## Screenshots
SCO:
image
SDO:
image

## Minimum version of Cortex XSOAR
- [x] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonTypes/IndicatorFields/indicatorfield-Country_Code.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "associatedToAll": true,
+    "associatedToAll": false,
     "caseInsensitive": true,
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-Country_Code.json:3]: Missing associatedTypes

File: Packs/CommonTypes/IndicatorFields/indicatorfield-Behavior.json [hunk 2]
```diff
@@ -1,55 +1,55 @@
...
- "sla": 0,
- "system": false,
- "threshold": 72,
- "type": "grid",
- "unmapped": false,
- "unsearchable": false,
- "useAsKpi": false,
- "version": -1,
- "fromVersion": "5.5.0"
+    "associatedToAll": false,
```
[the remark this answers is not in the document]
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-Behavior.json:2]: These all went trough demisto-sdk format, if the user will link it the system will add it again.

File: Packs/CommonTypes/IndicatorFields/indicatorfield-Country_Code.json (hunk 1 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-Country_Code.json:3]: same answer :)

File: Packs/CommonTypes/IndicatorFields/indicatorfield-Resource_Level.json [hunk 3]
```diff
@@ -40,6 +40,7 @@
     "sla": 0,
     "system": false,
     "systemAssociatedTypes": null,
+    "openEnded": true,
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-Resource_Level.json:43]: what is that?

File: Packs/CommonTypes/IndicatorFields/indicatorfield-dns_records.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            "required": false,
+            "script": "",
+            "selectValues": [
+                "TXT",
+                "SPF",
+                "MX",
+                "DMARC",
+                "SOA",
+                "TXT",
+                "CAA"
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-dns_records.json:25]: lets add 'A' and 'AAAA' records

File: Packs/CommonTypes/IndicatorFields/indicatorfield-positivedetections.json [hunk 5]
```diff
@@ -18,7 +18,7 @@
     "breachScript": "",
     "editForm": true,
     "description": "Number of engines that positively detected the indicator as malicious",
-    "associatedToAll": true,
+    "associatedToAll": false,
     "associatedTypes": null,
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-positivedetections.json:22]: Dont you think we should keep that field? (I know in the layout we have a section to see the verdict for each integration) but a field that we can use in playbooks and dashboards?

File: Packs/CommonTypes/IndicatorFields/indicatorfield-region.json [hunk 6]
```diff
@@ -24,13 +24,7 @@
...
     "group": 2,
     "hidden": false,
-    "associatedTypes": [
-        "IP",
-        "IPv6",
-        "CIDR",
-        "IPv6CIDR",
-        "STIX Threat Actor"
-    ],
+    "associatedTypes": [],
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-region.json:27]: What do you think with keeping the field for Threat Actor indicator type? For example, a group that operates from Russia, China, etc.'

File: Packs/CommonTypes/IndicatorFields/indicatorfield-signed.json [hunk 7]
```diff
@@ -19,10 +19,7 @@
     "editForm": true,
     "description": "",
     "associatedToAll": false,
-    "associatedTypes": [
-        "File",
-        "URL"
-    ],
+    "associatedTypes": [],
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-signed.json:22]: Don't you think we should keep the signatures data? maybe with a grid or different format but remove it from the file completely?

File: Packs/CommonTypes/IndicatorTypes/reputation-email.json [hunk 8]
```diff
@@ -1,75 +1,85 @@
...
-        ]
-      }
+    "id": "emailRep",
+    "version": -1,
+    "fromVersion": "5.0.0",
+    "modified": "2019-07-18T08:56:47.982384118Z",
+    "shouldPublish": false,
+    "shouldCommit": false,
+    "regex": "\\b[A-Za-z0-9.!#$%\u0026'*+/=?^_`{|}~\\p{L}-]{1,64}(@|\\[@\\])[A-Za-z0-9\\p{L}.-]{1,255}\\.[A-Za-z]{2,}\\b",
+    "details": "Email",
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorTypes/reputation-email.json:9]: The regex changed as well, planned? reason?

File: Packs/CommonTypes/ReleaseNotes/3_2_22.md [hunk 9]
```diff
@@ -1,4 +1,229 @@
 
+#### Indicator Fields
```
Comment (altmannyarden) [inline Packs/CommonTypes/ReleaseNotes/3_2_22.md:2]: we need only one release notes :)
I know that you need to update the RN every time, but lets have one file

File: Packs/CommonTypes/IndicatorFields/indicatorfield-positivedetections.json (hunk 5 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-positivedetections.json:22]: Leave it where? it should be an integration specific field (like in VT) what else do we use it for?

File: Packs/CommonTypes/IndicatorFields/indicatorfield-signed.json (hunk 7 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-signed.json:22]: We can, I truly have no opinion in the matter

File: Packs/CommonTypes/ReleaseNotes/3_2_22.md (hunk 9 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/ReleaseNotes/3_2_22.md]: 3_2_22 was a mistake, fixed it.

File: Packs/CommonTypes/IndicatorTypes/reputation-email.json (hunk 8 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorTypes/reputation-email.json:9]: It only encodes `&` which is later decoded:
image

File: Packs/CommonTypes/Layouts/layoutscontainer-url.json [hunk 10]
```diff
@@ -2,509 +2,470 @@
...
+                                "height": 44,
+                                "id": "75b654e0-9872-11ec-a272-434da3f30006",
+                                "index": 3,
+                                "sectionItemType": "field",
+                                "startCol": 0
+                            },
+                            {
+                                "endCol": 2,
+                                "fieldId": "creationdate",
+                                "height": 22,
```
Comment (altmannyarden) [inline Packs/CommonTypes/Layouts/layoutscontainer-url.json:131]: The field isn't associated to URL

File: Packs/CommonTypes/Layouts/layoutscontainer-url.json [hunk 11]
```diff
@@ -2,509 +2,470 @@
...
+                                "id": "0b9dd200-a09e-11e9-8956-390f602b039a",
+                                "index": 6,
+                                "listId": "e2c8c970-a09d-11e9-8956-390f602b039a",
+                                "sectionItemType": "field",
+                                "startCol": 0
+                            },
+                            {
+                                "endCol": 2,
+                                "fieldId": "techname",
```
[the remark this answers is not in the document]
Comment (altmannyarden) [reply Packs/CommonTypes/Layouts/layoutscontainer-url.json:159]: techname, techcountry, techorganization

File: Packs/CommonTypes/Layouts/layoutscontainer-url.json [hunk 12]
```diff
@@ -2,509 +2,470 @@
...
+                                "name": "Remove tags",
+                                "scriptId": "Builtin|||removeIndicatorField",
+                                "sectionItemType": "button",
+                                "startCol": 1
+                            }
+                        ],
+                        "maxW": 3,
+                        "minH": 1,
+                        "moved": false,
+                        "name": "Domain Details",
```
Comment (altmannyarden) [inline Packs/CommonTypes/Layouts/layoutscontainer-url.json:269]: should be URL details (this is URL layout)

**Comment by tschanfeld** [comment]
@altmannyarden @Ni-Knight You're only looking for a doc review of ReleaseNotes/3_2_26.md?

File: Packs/CommonTypes/Layouts/layoutscontainer-url.json (hunk 10 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/Layouts/layoutscontainer-url.json]: True

File: Packs/CommonTypes/Layouts/layoutscontainer-url.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (Ni-Knight) [reply Packs/CommonTypes/Layouts/layoutscontainer-url.json]: That is odd, removing them

File: Packs/CommonTypes/Layouts/layoutscontainer-url.json (hunk 12 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/Layouts/layoutscontainer-url.json]: True :)

File: Packs/CommonTypes/IndicatorFields/indicatorfield-dns_records.json (hunk 4 above)
Comment (altmannyarden) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-dns_records.json:25]: Im not sure if we will get those records as relationships , lets add them

File: Packs/CommonTypes/IndicatorFields/indicatorfield-signed.json (hunk 7 above)
Comment (altmannyarden) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-signed.json:22]: lets add it :)

File: Packs/CommonTypes/IndicatorFields/indicatorfield-associations.json [hunk 13]
```diff
@@ -26,7 +26,7 @@
     "hidden": false,
     "associatedTypes": null,
     "systemAssociatedTypes": null,
-    "associatedToAll": true,
+    "associatedToAll": false,
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-associations.json:29]: I saw URL and other indicator types using this field in their default mapping. What data returns usefully from this field?

File: Packs/CommonTypes/IndicatorTypes/reputation-Infrastructure.json [hunk 14]
```diff
@@ -32,9 +49,34 @@
         "stixid": {
             "simple": "Infrastructure.stixid",
             "complex": null
+        },
+        "tags": {
+            "simple": "",
+            "complex": {
+                "root": "File",
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorTypes/reputation-Infrastructure.json:56]: I think it should Infrastructure

File: Packs/CommonTypes/IndicatorFields/indicatorfield-positivedetections.json (hunk 5 above)
Comment (altmannyarden) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-positivedetections.json:22]: ok agree :)

File: Packs/CommonTypes/IndicatorTypes/reputation-ipv6.json [hunk 15]
```diff
@@ -22,150 +18,74 @@
...
-        },
-        "hostname": {
-            "simple": "",
-            "complex": {
-                "root": "IPv6",
-                "filters": [],
-                "accessor": "Hostname",
-                "transformers": []
-            }
+            "simple": "IP.Geo.Location",
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorTypes/reputation-ipv6.json:33]: shouldnt it be IPv6.geo. ...?

File: Packs/CommonTypes/IndicatorFields/indicatorfield-category.json [hunk 16]
```diff
@@ -19,11 +19,7 @@
     "editForm": true,
     "description": "",
     "associatedToAll": false,
-    "associatedTypes": [
-        "URL",
-        "Domain",
-        "DomainGlob"
-    ],
+    "associatedTypes": [],
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-category.json:22]: I think we should keep that field

File: Packs/CommonTypes/IndicatorTypes/reputation-cve.json [hunk 17]
```diff
@@ -3,164 +3,69 @@
...
-          "root": "CVE",
-          "filters": [],
-          "accessor": "TrafficLightProtocol",
-          "transformers": [{
-            "operator": "uniq",
-            "args": {}
-          }]
-        }
-      },
-      "tags": {
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorTypes/reputation-cve.json:150]: we dont want to keep it for the CVE indicator?

File: Packs/CommonTypes/IndicatorTypes/reputation-account.json [hunk 18]
```diff
@@ -3,220 +3,223 @@
...
-            {
-              "operator": "toUpperCase",
-              "args": {}
-            },
-            {
-              "operator": "uniq",
-              "args": {}
+        },
+        "indicatoridentification": {
+            "simple": "",
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorTypes/reputation-account.json:104]: indicatoridentification is not associated with that field

File: Packs/CommonTypes/IndicatorTypes/reputation-account.json [hunk 19]
```diff
@@ -3,220 +3,223 @@
...
-          "root": "Account",
-          "filters": [],
-          "accessor": "OrganizationUnit",
-          "transformers": [
-            {
-              "operator": "uniq",
-              "args": {}
+        },
+        "organizationalunitou": {
+            "simple": "",
```
Comment (altmannyarden) [inline Packs/CommonTypes/IndicatorTypes/reputation-account.json:122]: same here (no association to the field)

**Comment by tschanfeld** [comment]
made formatting changes to the release note file. if you need anything else from me for docs, let me know.

**Comment by altmannyarden** [comment]
@tschanfeld Please see and review the doc file also, which is the documentation we are planning to release with this PR

File: Packs/CommonTypes/IndicatorFields/indicatorfield-associations.json (hunk 13 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-associations.json:29]: Which field? `associations`? don't know, it's an old field

File: Packs/CommonTypes/IndicatorTypes/reputation-Infrastructure.json (hunk 14 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorTypes/reputation-Infrastructure.json:56]: And you are right :)

File: Packs/CommonTypes/IndicatorTypes/reputation-ipv6.json (hunk 15 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorTypes/reputation-ipv6.json:33]: it should, fixing.

File: Packs/CommonTypes/IndicatorFields/indicatorfield-category.json (hunk 16 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-category.json:22]: Why?

File: Packs/CommonTypes/IndicatorTypes/reputation-cve.json (hunk 17 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorTypes/reputation-cve.json:150]: Yarden, you need full comments, I can't follow 😄

Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorTypes/reputation-cve.json:150]: It isnt deleted, just moved up

File: Packs/CommonTypes/IndicatorTypes/reputation-account.json (hunk 18 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorTypes/reputation-account.json]: Removed

File: Packs/CommonTypes/IndicatorTypes/reputation-account.json (hunk 19 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorTypes/reputation-account.json]: Removed

File: Packs/CommonTypes/IndicatorFields/indicatorfield-signed.json (hunk 7 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-signed.json:22]: Do you have examples for that data?

**Comment by tschanfeld** [comment]
@altmannyarden Can you add the file to the PR? I only see the one markdown file for the release note.

**Comment by altmannyarden** [comment]
@tschanfeld In the description....
Screen Shot 2022-03-16 at 14 58 26

**Comment by tschanfeld** [comment]
@altmannyarden This doc is going in xsoar.pan.dev? Is this only related to certain versions of Cortex XSOAR?

File: Packs/CommonTypes/IndicatorFields/indicatorfield-CVSS_Score.json [hunk 20]
```diff
@@ -17,12 +17,13 @@
...
     "hidden": false,
+    "openEnded": false,
     "associatedTypes": [
-        "Infrastructure"
+        "CVE"
     ],
     "associatedToAll": false,
     "unmapped": false,
-    "unsearchable": false,
+    "unsearchable": true,
```
Comment (idovandijk) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-CVSS_Score.json:26]: Fields' "unsearchable" key cannot be modified, existing customers will keep previous behavior unless they explicitly uninstall and then install the pack from scratch

File: Packs/CommonTypes/IndicatorFields/indicatorfield-Behavior.json (hunk 2 above)
Comment (idovandijk) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-Behavior.json:2]: comma (move it to fromVersion)

File: Packs/CommonTypes/IndicatorFields/indicatorfield-External_References.json [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+    "required": false,
+    "sla": 0,
+    "system": false,
+    "threshold": 72,
+    "type": "grid",
+    "unmapped": false,
+    "unsearchable": true,
+    "useAsKpi": false,
+    "version": -1,
+    "fromVersion": "5.5.0"
```
Comment (idovandijk) [inline Packs/CommonTypes/IndicatorFields/indicatorfield-External_References.json:58]: IIRC we are no longer releasing content from 5.5, unless this case is an exception. I think they should all be 6.0.0 or 6.1.0

File: Packs/CommonTypes/IndicatorFields/indicatorfield-CVSS_Score.json (hunk 20 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-CVSS_Score.json:26]: It is known, an also true to all fields values.

File: Packs/CommonTypes/IndicatorFields/indicatorfield-External_References.json (hunk 21 above)
Comment (Ni-Knight) [reply Packs/CommonTypes/IndicatorFields/indicatorfield-External_References.json]: Updated

**Pull request closed without merging**